src/passes/pass.cpp

```diff
@@ -192,7 +193,7 @@ void PassRunner::run() {
       std::cerr << "final module does not validate\n";
       abort();
     }
-    passDebug = currPassDebug;
+    recursion--;
```

Comment:
Hm, that's two good points. Ok, I'm fine with merging this as-is for now. I'm sure there's a way to refactor this to better isolate the debug weirdness from the rest of the code, but my ideas at the moment are nontrivial.